src/components/MissionAPP/MissionShelf.jsx

```diff
@@ -0,0 +1,49 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { useEffect } from 'react';
+import { getMissions, selectMission } from 'features/missions/missionsSlice';
+import { nanoid } from '@reduxjs/toolkit';
+
+const MissionShelf = () => {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(getMissions());
+  }, []);
+
+  const { missions, isLoading, error } = useSelector(selectMission);
+
+  const nobooks = (
+    <h3 className="center">No Books Found</h3>
+  );
+
+  if (isLoading) return <h3 className="center">Loading...</h3>;
+  if (error) {
+    return (
+      <h3 className="center">
+        Error:
+        {' '}
+        {error}
+      </h3>
+    );
+  }
+  if (missions.length < 1) return <>{nobooks}</>;
```

Comment:
- [ ] For the same reason mentioned above here we should have `{noMission}`.

Reply:
Got catch about the names. Already changed.